### PR TITLE
adding my setuptools-github project recipe

### PR DESCRIPTION
**Pull request by cav71**
Hi,
this is my conda recipe for a package I developed. Can you please help with the conda-forge release?
Thanks

@conda-forge-admin, please ping team

### CODE REVIEW

**Comment by isuruf** [comment]
Why are you overriding `setuptools/__init__.py`?

**Comment by cav71** [comment]
> Why are you overriding `setuptools/__init__.py`?

It is needed only for internal tests. In fact in setup.py:
> packages = find_namespace_packages(where="src")
> packages.remove("setuptools")

This will *NOT* overrides setuptools. 

Unfirtunately this is necessary because the setuptools-github lives under setuptools tree, but setuptools adds code in the __init__.py (so I cannot use the standard package extension mechanism).

File: recipes/setuptools-github/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  sha256: bf5ebf0ad81a872a9bb50123ff33387f82d5c80467949115cf06eca003479670
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
Comment (isuruf) [inline recipes/setuptools-github/meta.yaml:19]: Why do you need a compiler?

File: recipes/setuptools-github/LICENSE [hunk 2]
```diff
@@ -0,0 +1,25 @@
+BSD 2-Clause License
```
Comment (isuruf) [inline recipes/setuptools-github/LICENSE:1]: This should be taken from the tarball. Can you remove this file?

File: recipes/setuptools-github/meta.yaml (hunk 1 above)
Comment (cav71) [reply recipes/setuptools-github/meta.yaml]: removing..

File: recipes/setuptools-github/LICENSE (hunk 2 above)
Comment (cav71) [reply recipes/setuptools-github/LICENSE]: done

**Comment by cav71** [comment]
@isuruf is there anything else I can do for this recipe?

File: recipes/setuptools-github/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  host:
+    - python >=3.6
+    - pip
+  run:
+    - python >=3.6
+
+# test:
+#   requires:
+#     - pip
+#     - pytest
```
Comment (dopplershift) [inline recipes/setuptools-github/meta.yaml:27]: We generally need some kind of test to validate that the created conda package isn't broken. Is there some kind of `import` we can do here or a command to run?

Comment (cav71) [reply recipes/setuptools-github/meta.yaml]: done

File: recipes/setuptools-github/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python >=3.6
+    - pip
+  run:
+    - python >=3.6
```
Comment (dopplershift) [inline recipes/setuptools-github/meta.yaml:22]: As a setuptools plugin, this needs `setuptools` installed to function properly, right? In that case you should have it here:
```suggestion
    - python >=3.6
    - setuptools
```

**Review by dopplershift — APPROVED** [review]

**Pull request merged**